toptal/picasso
# feat(TreeNodeAvatar): add objectFit prop support for TreeView/TreeNodeAvatar

[TOP-1879]

### Description

TreeNodeAvatar always renders a square container for the image, though when image itself is not squared, UI might feel like there are undesired margins. `objectFit=cover` lets client code always render an actual squared picture no matter what's the aspect ratio of source image.


### How to test

- Ensure there's no change in how existing TreeView instances render avatars on their nodes

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| <img width="213" alt="image" src="https://user-images.githubusercontent.com/903437/147296593-a9b6a57c-65eb-45f1-b2b2-309da392976d.png"> | <img width="215" alt="image" src="https://user-images.githubusercontent.com/903437/147296725-f253b5cd-09a1-45d2-874f-e68c59bf46f3.png"> |

### Review


<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and
1. You have no extra requests.
2. You have optional requests.
   1. Add `nit: ` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because
1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:
1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[TOP-1879]: https://toptal-core.atlassian.net/browse/TOP-1879?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
